src/backend/query_builder.rs

```diff
@@ -1378,29 +1343,49 @@ pub trait QueryBuilder: QuotedBuilder + EscapeBuilder + TableRefBuilder {
         right: &SimpleExpr,
         sql: &mut dyn SqlWriter,
     ) {
-        let no_paren = matches!(op, BinOper::Equal | BinOper::NotEqual);
-        let left_paren = left.need_parentheses()
-            && left.is_binary()
-            && op != left.get_bin_oper().unwrap()
-            && !no_paren;
+        // If left has higher precedence than op, we can drop parentheses around left
+        let drop_left_higher_precedence =
+            self.inner_expr_well_known_greater_precedence(left, &(*op).into());
+
+        //Figure out if left associativity rules allow us to drop the left parenthesis
```

Comment:
Super-nit: Space at the beginning of the comment. Feel free to ignore.

Reply:
Fixed :-)